[WebGPU] Remove call to entryPoint now that we don't need to transform them anymore

LumioseSil opened the pull request:
#### 36df45bed29e7f61941f8b1c5bae8916b03d0c52
<pre>
[WebGPU] Remove call to entryPoint now that we don&apos;t need to transform them anymore
<a href="https://bugs.webkit.org/show_bug.cgi?id=279470">https://bugs.webkit.org/show_bug.cgi?id=279470</a>

Reviewed by Mike Wyrzykowski.

We can handle unicode now, so any entry point
transformation code can be removed.

* Source/WebGPU/WebGPU/Pipeline.mm:
  (WebGPU::createLibrary): Remove call to transformedEntryPoint.
* Source/WebGPU/WebGPU/ShaderModule.h:
  (WebGPU::ShaderModule::library const): Ditto.
* Source/WebGPU/WebGPU/ShaderModule.mm:
  (WebGPU::ShaderModule::transformedEntryPoint const): Deleted.

Canonical link: <a href="https://commits.webkit.org/283466@main">https://commits.webkit.org/283466@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/23d481ad0a48dc998e4c54e738263a7bee18e478

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66362 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45737 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18983 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70395 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16973 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53536 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17255 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53224 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11841 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69429 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42164 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57445 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33877 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38835 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15848 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60731 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15176 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72098 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10319 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14552 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60547 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10351 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57515 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60859 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14629 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8515 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2131 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41544 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42621 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43804 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42364 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebGPU/WebGPU/Pipeline.mm:
```diff
@@ -88,7 +86,6 @@
         auto keyEntry = fromAPI(entry.key);
         auto indexIterator = entryPointInformation.specializationConstants.find(keyEntry);
-        // FIXME: Remove code inside the following conditional statement when https://bugs.webkit.org/show_bug.cgi?id=266774 is completed
         if (indexIterator == entryPointInformation.specializationConstants.end()) {
```
I think we can remove lines 89..100 as well, this was to workaround the prior lack of unicode support but is no longer needed

mwyrzykowski replied:
Well to be more specific, it should be replaced with (from lines 99-100):
```
if (indexIterator == entryPointInformation.specializationConstants.end())
    return { };
```

mwyrzykowski reviewed (commented):
This is great, thank you for removing this effectively unused code.

Only suggestion is to remove the block under the conditional, replacing it with:
```
if (indexIterator == entryPointInformation.specializationConstants.end())
    return { };
```

LumioseSil committed:
[WebGPU] Remove call to entryPoint now that we don't need to transform them anymore
https://bugs.webkit.org/show_bug.cgi?id=279470

Reviewed by Mike Wyrzykowski.

We can handle unicode now, so any entry point
transformation code can be removed.

* Source/WebGPU/WebGPU/Pipeline.mm:
  (WebGPU::createLibrary): Remove call to transformedEntryPoint.
* Source/WebGPU/WebGPU/ShaderModule.h:
  (WebGPU::ShaderModule::library const): Ditto.
* Source/WebGPU/WebGPU/ShaderModule.mm:
  (WebGPU::ShaderModule::transformedEntryPoint const): Deleted.

Canonical link: https://commits.webkit.org/283466@main
